src/Gui.rs

```diff
@@ -51,14 +51,13 @@ impl Gui{
     }
 
     pub fn update(&mut self) -> bool{
-        for event in self.events.poll_iter(){
-            match event{
-                Event::Quit {..} => {return false;},
-                _ => {}
-            }
+        if self.events.poll_iter().any(|i| matches!(i, Event::Quit {..})) {
+            false
+        }
+        else {
+            self.canvas.present();
+            true
```

Comment:
I wish there was "return" but since it's rust it's not here (tbh it's better this way but C is still fresh in my mind)

Reply:
Yes i agree but Rust all the way